plugwise/smile.py

```diff
@@ -31,59 +35,57 @@
         """Helper-function for smile.py: _all_device_data().
         When conditions are met, the plugwise_notification binary_sensor is appended.
         """
         if d_id == self.gateway_id:
-            if self._sm_thermostat is not None:
+            if self._is_thermostat:
                 bs_dict.update(PW_NOTIFICATION)
 
-    def _all_device_data(self):
+    def _all_device_data(self) -> None:
         """Helper-function for get_all_devices().
         Collect initial data for each device and add to self.gw_data and self.gw_devices.
         """
-        for dev_id, dev_dict in self._devices.items():
-            dev_and_data = dev_dict
-            temp_bs_dict = {}
-            temp_s_dict = {}
-            temp_sw_dict = {}
-            data = self._get_device_data(dev_id)
+        for device_id, device in self._devices.items():
+            temp_bs_dict: dict[str, bool] = {}
+            temp_s_dict: dict[str, Any] = {}
+            temp_sw_dict: dict[str, bool] = {}
+            data: dict[str, Any] = self._get_device_data(device_id)
 
             self._create_dicts_from_data(data, temp_bs_dict, temp_s_dict, temp_sw_dict)
-            self._append_special(data, dev_id, temp_bs_dict, temp_s_dict)
-
-            dev_and_data.update(data)
+            self._append_special(device_id, temp_bs_dict, temp_s_dict)
+            device.update(data)
             if temp_bs_dict:
-                dev_and_data["binary_sensors"] = temp_bs_dict
+                device["binary_sensors"] = temp_bs_dict
             if temp_s_dict:
-                dev_and_data["sensors"] = temp_s_dict
+                device["sensors"] = temp_s_dict
             if temp_sw_dict:
-                dev_and_data["switches"] = temp_sw_dict
+                device["switches"] = temp_sw_dict
 
-            self.gw_devices[dev_id] = dev_and_data
+            self.gw_devices[device_id] = device
 
-        self.gw_data["active_device"] = self._ot_device or self._on_off_device
+        self.gw_data["active_device"] = self._opentherm_device or self._on_off_device
         self.gw_data["cooling_present"] = self._cooling_present
         self.gw_data["gateway_id"] = self.gateway_id
         self.gw_data["heater_id"] = self._heater_id
-        self.gw_data["single_master_thermostat"] = self._sm_thermostat
+        self.gw_data["single_master_thermostat"] = (
+            self._is_thermostat and not self._multi_thermostats
+        )
         self.gw_data["smile_name"] = self.smile_name
 
-    def get_all_devices(self):
+    def get_all_devices(self) -> None:
         """Determine the devices present from the obtained XML-data."""
-        self._devices = {}
+        self._devices: dict[str, dict[str, Any]] = {}
         self._scan_thermostats()
         self.single_master_thermostat()
 
         for appliance, details in self._appl_data.items():
-            loc_id = details["location"]
+            loc_id: str | None = details["location"]
```

Comment:
Instead of adding local typing, type whatever provides _appl_data  (with a TypeDict).

Reply:
Ok, yes `details` is part of the input, removed.